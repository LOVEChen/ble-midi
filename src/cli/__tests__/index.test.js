/* eslint-env jest */
jest.mock('yargs');

test('handles help', () => {
  const lib = require('yargs');
  const cli = require('../index').default;
  cli([]);
  const mockYargs = lib.getMockYargs();
  expect(mockYargs.usage.mock).toMatchSnapshot();
  expect(mockYargs.commandDir.mock).toMatchSnapshot();
  expect(mockYargs.help.mock).toMatchSnapshot();
  expect(mockYargs.alias.mock).toMatchSnapshot();
  expect(mockYargs.version.mock).toMatchSnapshot();
  expect(mockYargs.demand.mock).toMatchSnapshot();
  expect(mockYargs.strict.mock).toMatchSnapshot();
});
