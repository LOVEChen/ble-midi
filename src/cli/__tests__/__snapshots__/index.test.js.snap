// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles help 1`] = `
Object {
  "calls": Array [
    Array [
      "Usage: $0 <command> [options]",
    ],
  ],
  "instances": Array [
    [Function],
  ],
}
`;

exports[`handles help 2`] = `
Object {
  "calls": Array [
    Array [
      "cmds",
    ],
  ],
  "instances": Array [
    [Function],
  ],
}
`;

exports[`handles help 3`] = `
Object {
  "calls": Array [
    Array [
      "help",
    ],
  ],
  "instances": Array [
    [Function],
  ],
}
`;

exports[`handles help 4`] = `
Object {
  "calls": Array [
    Array [
      "h",
      "help",
    ],
    Array [
      "v",
      "version",
    ],
  ],
  "instances": Array [
    [Function],
    [Function],
  ],
}
`;

exports[`handles help 5`] = `
Object {
  "calls": Array [
    Array [
      "ble-midi version 0.0.1",
    ],
  ],
  "instances": Array [
    [Function],
  ],
}
`;

exports[`handles help 6`] = `
Object {
  "calls": Array [
    Array [
      1,
    ],
  ],
  "instances": Array [
    [Function],
  ],
}
`;

exports[`handles help 7`] = `
Object {
  "calls": Array [
    Array [],
  ],
  "instances": Array [
    [Function],
  ],
}
`;
